commons-vfs2-sandbox/src/main/java/org/apache/commons/vfs2/provider/smb2/Smb2ClientWrapper.java

```diff
@@ -0,0 +1,223 @@
+
+        this.root = root;
+        this.fileSystemOptions = fileSystemOptions;
+        smbClient = new SMBClient(CONFIG);
+        setupClient();
+    }
+
+    private void setupClient() throws FileSystemException {
+
+        final GenericFileName rootName = getRoot();
+
+        //the relevant data to authenticate a connection
+        String userName;
+        if (rootName.getUserName().equals("") || rootName.getUserName() == null) {
+            userName = "";
+        } else {
+            if (rootName.getUserName().contains(";")) {
+                userName = rootName.getUserName().substring(rootName.getUserName().indexOf(";") + 1, rootName.getUserName().length());
+            } else {
+                userName = rootName.getUserName();
+            }
+        }
+        String password = rootName.getPassword();
+        String authDomain = (rootName.getUserName().contains(";") ?
+                rootName.getUserName().substring(0, rootName.getUserName().indexOf(";")) : null);
+
+        //if username == "" the client tries to authenticate "anonymously". It's also possible to submit "guest" as username
+        AuthenticationContext authContext = new AuthenticationContext(userName, password.toCharArray(), authDomain);
+
+        //a connection stack is: SMBClient > Connection > Session > DiskShare
+        try {
+            connection = smbClient.connect(rootName.getHostName());
+            session = connection.authenticate(authContext);
+            String share = ((Smb2FileName) rootName).getShareName();
+            diskShare = (DiskShare) session.connectShare(share);
+        } catch (Exception e) {
+            throw new FileSystemException("vfs.provider.smb2/connect.error", rootName.getHostName(), e.getCause());
+        }
+    }
+
+    public GenericFileName getRoot() {
+
+        return root;
+    }
+
+    /**
+     * Returns file information of a given file.
+     *
+     * @param relativePath of file
+     * @return FileAllInformation object
+     */
+    public FileAllInformation getFileInfo(String relativePath) {
+
+        try {
+            return diskShare.getFileInformation(relativePath);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    /**
+     * Returns a write handle on the file.
+     *
+     * @param path path to file
+     * @return DiskEntry file
+     */
+    public DiskEntry getDiskEntryWrite(String path) {
+
+        return diskShare.open(path,
+                EnumSet.of(AccessMask.MAXIMUM_ALLOWED),
+                EnumSet.of(FileAttributes.FILE_ATTRIBUTE_NORMAL),
+                EnumSet.of(SMB2ShareAccess.FILE_SHARE_WRITE),
+                SMB2CreateDisposition.FILE_OPEN_IF,
+                EnumSet.of(SMB2CreateOptions.FILE_NO_COMPRESSION));
+    }
+
+    /**
+     * Creates a folder on a given path.
+     *
+     * @param path of folder
+     */
+    public void createFolder(String path) {
+
+        DiskEntry de = getDiskEntryFolderWrite(path);
+        de.close();
```

Comment:
let's close this in a finally block?

Reply:
no need to use a finally block here right?